### ISSUE TITLE
use `xtal2png` with `imagen-pytorch` and `matbench-genmetrics`

### ISSUE THREAD

**Issue by sgbaird** [sparks-baird/xtal2png#204]
`matbench-genmetrics` is in a usable state now https://github.com/sparks-baird/xtal2png/issues/12#issuecomment-1221248835

I think `imagen-pytorch` can be used with TPU, but I'm not sure how much custom configuration is required https://github.com/sparks-baird/xtal2png/blob/main/notebooks/3.1-imagen-pytorch.ipynb

I might just need to try it on Colab, switch to TPU, and see what happens. I think the latest versions uses 🤗 Accelerate library will make it easier to switch over. I'm unsure if I should focus more on hyperparameter tuning or just pick some reasonable defaults and train it for as long as seems reasonable (a week or two, for example). If going with my university HPC instead of TPU time, I can still do checkpointing in either case.

**Comment by sgbaird** [comment]
@ ~2000 epochs (4x4 tile)
![image](https://user-images.githubusercontent.com/45469701/197320308-335354a7-1390-489c-9cf0-96da593fd142.png)

**Comment by kjappelbaum** [comment]
>@ ~2000 epochs (4x4 tile)

do they decode to some reasonable materials? :D

**Comment by sgbaird** [comment]
> do they decode to some reasonable materials? :D

I'm going to go with a pretty confident "no" 😬 

![xtal2png-imagen-pytorch-epoch=1999-6x5](https://user-images.githubusercontent.com/45469701/198710773-95da754c-251d-4c16-b0fe-a3c4792003f6.png)

I think I'm also going to say the metrics need some work (note this is for 1000 generated structures):
```python
{0: {'validity': 0.4092998941577499, 'coverage': 0.0, 'novelty': 1.0, 'uniqueness': 1.0}}
```

**Comment by sgbaird** [comment]
While I'm sure there's a lot to be done with the hyperparameters, I think I'll take another shot at running CDVAE for comparison.

**Comment by HarshaSatyavardhan** [comment]
I am seeing coverage as 0 is it not concerning?. I my self have tried and got coverage as 0 with different variations of ddpm.

**Comment by sgbaird** [comment]
@HarshaSatyavardhan thanks for the great question. Concerning - yes, though the idea of rediscovery is quite difficult. To make the point, see what the authors of [PGCGM](https://github.com/MilesZhao/PGCGM) needed to do before "moving the needle" past 0 in https://www.nature.com/articles/s41524-023-01059-8:

![image](https://github.com/sparks-baird/xtal2png/assets/45469701/774af1ed-201f-4267-b80f-978c964699a6)

Notice how the first bar along the horizontal axis starts at 50*10000 = 500,000. Also, the coverage benchmark from matbench-genmetrics is even more difficult to succeed at than what PGCGM did because it uses time-based splits (i.e., not just can we discover something that was held out, but can we discover something "in the future" based only on training data before some calendar year).

> I my self have tried and got coverage as 0 with different variations of ddpm.

Thank you for sharing this.

Open to thoughts or suggestions you have. I think both xtal2png and the benchmarks themselves could be improved. Aside: matbench-genmetrics is under review at https://github.com/openjournals/joss-reviews/issues/5618.

cc @hasan-sayeed @sp8rks @michaeldalverson
